bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/PerChannelBookieClient.java

```diff
@@ -732,14 +729,18 @@ private void readEntryInternal(final long ledgerId,
                     .build();
         }
 
-        CompletionValue completion = new ReadCompletion(completionKey, cb,
-                                                        ctx, ledgerId, entryId);
-        CompletionValue existingValue = completionObjects.putIfAbsent(
-                completionKey, completion);
+        ReadCompletion readCompletion = new ReadCompletion(completionKey, cb, ctx, ledgerId, entryId);
+        CompletionValue existingValue = completionObjects.putIfAbsent(completionKey, readCompletion);
         if (existingValue != null) {
-            // There's a pending read request on same ledger/entry. This is not supported in V2 protocol
-            LOG.warn("Failing concurrent request to read at ledger: {} entry: {}", ledgerId, entryId);
-            completion.errorOut(BKException.Code.UnexpectedConditionException);
+            // There's a pending read request on same ledger/entry. Use the multimap to track all of them
+            synchronized (this) {
+                completionObjectsV2Conflicts.put(completionKey, readCompletion);
+            }
+        }
+
+        final Channel c = channel;
```

Comment:
This is handled in writeAndFlush now. You'll need to add a check for the conflicts map into the #errorOut method.

Reply:
Ok, I've missed that part. I'll fix that